File: BeachMapMobile/BeachMapMobile/Models/Projeto.swift
//
//  File.swift
//  BeachMapMobile
//
//  Created by Wilton Garcia on 26/08/20.
//  Copyright © 2020 LAVitor Pires. All rights reserved.
//

import Foundation

struct Projeto: Codable{
    
    let id: Int
    let nome: String
    let escritorio: String
    let descricao: String
    let desafios: String
    let techStack: String
    let oportunidade: [String]
    let status: String
    let dataUltimaAtualizacao: String
    let responsaveis: String
    
    init(id:Int, nome:String, escritorio:String, descricao:String, responsaveis:String, desafios:String, techStack:String, oportunidades:[String], status:String, dataUltimaAtualizacao:String){
       
        self.id = id
        self.nome = nome
        self.escritorio = escritorio
        self.descricao = descricao
        self.responsaveis = responsaveis
        self.desafios = desafios
        self.techStack = techStack
        self.oportunidade = oportunidades
        self.status = status
        self.dataUltimaAtualizacao = dataUltimaAtualizacao
        
        
    }
    
}

File: BeachMapAPI/Modelos/Projeto.py
from pydantic import BaseModel, constr
from typing import List

class Projeto(BaseModel):
    id: int
    nome: constr(min_length=3, max_length=100)
    escritorio: str
    status: str
    dataUltimaAtualizacao: str
    descricao: constr(min_length=5, max_length=4000)
    desafios: str
    techStack: str
    oportunidade: List[str]
    responsaveis: str